modules/replicator/src/main/java/org/apache/ignite/internal/replicator/Replica.java

```diff
@@ -119,7 +117,7 @@ public Replica(
         this.replicaGrpId = replicaGrpId;
         this.listener = listener;
         this.storageIndexTracker = storageIndexTracker;
-        this.raftClient = raftClient;
+        this.raftClient = raftClient();
```

Comment:
Why not simple listener.raftClient()?

Reply:
✅ Just because of class casting: inside the method I cast client to the specific `Replica`'s raft client type. And a better way to localize such too specific (workaround? see TODO there) things in one certain method.